library/math/numbersLine/src/components/Jump.tsx

```diff
@@ -3,72 +3,65 @@ import jumpArrowMinus from "/assets/icons/jumpArrowMinus.png";
 import React, { useEffect, useState } from "react";
 import { IElement } from "../type/moveable";
 import MoveableElement from "./MoveableElement";
-import { useNumbersLineContext } from "@/context/numbersLineContext";
+import { useNumbersLineContext } from "../context/numbersLineContext";
 import { calculatRulerWidth, calculatUnitsAmount } from "../lib/utils";
 import { RulerPadding, jumpArrowHeight } from "../consts/elementConsts";
-import { MatchBaseJumpClassName, calcHeightStartPosition, calcWidthStartPosition } from "../lib/stylesUtils";
+import { MatchBaseJumpClassName } from "../lib/stylesUtils";
+import { LineRange } from "@/type/ruler";
 
 interface IProps {
   element: IElement;
 }
 
 const Jump = ({ element }: IProps) => {
-  const { windowSize, type, dragElements, setDragElements, idDraggElementClick } = useNumbersLineContext();
-  const [unit, setUnit] = useState(windowSize.width / calculatUnitsAmount(type));
-  const [isJumpUnderRuler, setIsJumpUnderRuler] = useState(false);
+  const { windowSize, typeRuler, dragElements, idDraggElementClick } = useNumbersLineContext();
+  const [unit, setUnit] = useState(windowSize.width / calculatUnitsAmount(typeRuler));
+  const [hideNumber, setHideNumber] = useState(true);
   const moveableRef = React.useRef<HTMLDivElement>(null);
 
   useEffect(() => {
-    let rulerWidth = calculatRulerWidth(windowSize.width, RulerPadding) / calculatUnitsAmount(type);
+    let rulerWidth = calculatRulerWidth(windowSize.width, RulerPadding) / calculatUnitsAmount(typeRuler);
     setUnit(rulerWidth);
-  }, [type, windowSize.width]);
+  }, [typeRuler, windowSize.width]);
 
   useEffect(() => {
-    const root = document.documentElement;
-    root.style.setProperty("--margin-value", `${isJumpUnderRuler ? -50 : 28}px`);
-  }, [isJumpUnderRuler]);
-
-  const changeHidenumbers = () => {
-    let newElements = dragElements.map((item: IElement) => (item.id === idDraggElementClick ? { ...item, hideNumber: !item.hideNumber } : item));
-    setDragElements(newElements);
-  };
+    dragElements.forEach((item) => {
+      const element = document.getElementById("dragElement-jump" + item.id);
+      if (element && item.transform != element.style.transform) {
+        element.style.transform = item.transform;
+        element.style.width = `${item.value * unit}px`;
+      }
+      const root = document.documentElement;
+      root.style.setProperty("--margin-value", `${item.underRuler ? -50 : 28}px`);
+    });
+  }, [dragElements]);
```

Comment:
why cant you change the style when you render the element. style={{transform:item.transform etc...}}
more normal